app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,9 @@
+class WelcomeController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
@jw9billions change this to: 

`before_action :authenticate_user!, except: [:index, :about]`

so that non-signed user can see your welcome pages. I did this locally and am getting: "RuntimeError in Welcome#index" You need to check if there is a signed in user around this line: 

```
<%= link_to "Sign up", new_registration_path(current_user) %>
```